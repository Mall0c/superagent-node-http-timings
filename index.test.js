const request = require('superagent');
const logNetworkTime = require('./index');
jest.setTimeout(10000);
describe('GIVEN plugin was used WHEN response arrives THEN callback should get timings information', () => {
  test('timings', () => {
    const assertions = [
      'verified that timings were populated'
    ];
    expect.assertions(assertions.length);
    return expect(new Promise((res, rej) => {
      request
      .get(`https://google.com`)
      .use(logNetworkTime((err, result) => {
        if (err) {
          return rej(err);
        }
        res(valuesToTypes(result));
      }))
      .then(x => x)
      .catch(rej);
    })).resolves.toMatchSnapshot();
  });
});

function valuesToTypes(obj) {
  return Object.keys(obj).reduce((acc, key) => {
    const value = obj[key];
    const type = typeof value;
    acc[key] = type === 'object' ? valuesToTypes(value) : type;
    return acc;
  }, {})
}