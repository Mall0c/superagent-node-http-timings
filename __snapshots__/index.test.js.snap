// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN plugin was used WHEN response arrives THEN callback should get timings information timings 1`] = `
Object {
  "status": "number",
  "timings": Object {
    "contentTransfer": "number",
    "dnsLookup": "number",
    "firstByte": "number",
    "socketAssigned": "number",
    "tcpConnection": "number",
    "tlsHandshake": "number",
    "total": "number",
  },
  "url": "string",
}
`;
